pkg/manifests/manifests.go

```diff
@@ -0,0 +1,193 @@
+package manifests
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/http"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+func configureManifests(ctx *image.Context) error {
+	var downloadedManifests []string
+	var combinedManifestPaths []string
+	var extractedImages []string
+	var err error // created here to avoid scoping issues with "downloadedManifests" when using := on line 29
+
+	if len(ctx.ImageDefinition.Kubernetes.Manifests.URLs) != 0 {
+		downloadDestination := filepath.Join(ctx.CombustionDir, "downloaded-manifests")
+		if err := os.MkdirAll(downloadDestination, os.ModePerm); err != nil {
+			return fmt.Errorf("creating %s dir: %w", downloadDestination, err)
+		}
+
+		downloadedManifests, err = downloadManifests(ctx, downloadDestination)
+		if err != nil {
+			return fmt.Errorf("error downloading manifests: %w", err)
+		}
+	}
+
+	localManifestSrcDir := filepath.Join(ctx.ImageConfigDir, "manifests")
+	localManifestDestDir := filepath.Join(ctx.ImageConfigDir, "local-manifests")
+	copiedManifests, err := copyManifests(localManifestSrcDir, localManifestDestDir)
+	if err != nil {
+		return fmt.Errorf("error copying manifests: %w", err)
+	}
+
+	combinedManifestPaths = append(copiedManifests, downloadedManifests...)
+
+	for _, manifestPath := range combinedManifestPaths {
+		manifestData, err := readManifest(manifestPath)
+		if err != nil {
+			return fmt.Errorf("error reading manifest %w", err)
+		}
+
+		foundImages, err := findImagesInManifest(manifestData)
+		if err != nil {
+			return fmt.Errorf("error finding images in manifest %w", err)
+		}
+		extractedImages = append(extractedImages, foundImages...)
+	}
+
+	extractedImages = removeDuplicateImages(extractedImages)
```

Comment:
Seems obsolete if `addImagesToRegistryDefinition` already has a check for it. `findImagesInManifest` also lists the images in a map first and then goes through a slice and then through a map again. Can we reuse a data structure?

Reply:
I've restructured it to be a much cleaner flow